pallets/order-book/src/lib.rs

```diff
@@ -424,39 +428,22 @@ pub mod pallet {
 		pub fn user_update_order(
 			origin: OriginFor<T>,
 			order_id: T::OrderIdNonce,
-			buy_amount: T::Balance,
-			price: T::SellRatio,
+			amount_out: T::Balance,
+			price: OrderPrice<T::SellRatio>,
 		) -> DispatchResult {
 			let account_id = ensure_signed(origin)?;
 			let order = Orders::<T>::get(order_id)?;
-			let min_fulfillment_amount = T::DecimalConverter::to_asset_balance(
-				T::MinFulfillmentAmountNative::get(),
-				order.asset_in_id,
-			)?;
+
+			ensure!(
+				account_id == order.placing_account,
+				Error::<T>::Unauthorised
+			);
 
 			Self::inner_update_order(
-				account_id.clone(),
-				order_id,
-				buy_amount,
+				order.clone(),
+				amount_out,
 				price,
-				min_fulfillment_amount,
-				|order| {
-					ensure!(
-						account_id == order.placing_account,
-						Error::<T>::Unauthorised
-					);
-
-					let min_amount =
-						TradingPair::<T>::get(&order.asset_in_id, &order.asset_out_id)?;
-					Self::is_valid_order(
-						order.asset_in_id,
-						order.asset_out_id,
-						order.buy_amount,
-						order.max_sell_rate,
-						order.min_fulfillment_amount,
-						min_amount,
-					)
-				},
+				TradingPair::<T>::get(&order.asset_in_id, &order.asset_out_id)?,
```

Comment:
We are defaulting the `min_fullfillment_amount` here to the `min_order_amount`, right? That is your idea?

Reply:
No, I changed the `unner_updat_order()` method. `min_fulfillment_amount` is got from the inside of the function, and `min_order_amount` is passed from the outside.

There is no change in the logic of `min_fulfillment_amount` or `min_order_amount`; I just moved some things around to simplify the validation of these methods.

Reply:
Cool